arborist/aws_client.go

```diff
@@ -0,0 +1,142 @@
+package arborist
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+// S3Credentials contains AWS credentials
+type S3Credentials struct {
+	region             string
+	awsAccessKeyID     string
+	awsSecretAccessKey string
+}
+
+type AwsClient struct {
+	credentials S3Credentials
+	session     *session.Session
+	bucket      string
+}
+
+func (client *AwsClient) SetS3Credentials(cred S3Credentials) {
+	client.credentials = cred
+}
+
+// LoadCredentialFromConfigFile loads AWS credentials from the config file
+func (client *AwsClient) LoadCredentialFromConfigFile(path string) error {
+	// Read data file
+	jsonBytes, err := ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	// Get AWS region
+	data, err := GetValueFromKeys(jsonBytes, []string{"AWS", "region"})
+	if err != nil {
+		return err
+	}
+	client.credentials.region = data.(string)
+
+	// Get AWS access key id
+	data, err = GetValueFromKeys(jsonBytes, []string{"AWS", "aws_access_key_id"})
+	if err != nil {
+		return err
+	}
+	client.credentials.awsAccessKeyID = data.(string)
+
+	// Get AWS secret access key
+	data, err = GetValueFromKeys(jsonBytes, []string{"AWS", "aws_secret_access_key"})
+	if err != nil {
+		return err
+	}
+	client.credentials.awsSecretAccessKey = data.(string)
+
+	return nil
+}
+
+// createNewSession creats a aws s3 session
+func (client *AwsClient) createNewSession() error {
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(client.credentials.region),
+		Credentials: credentials.NewStaticCredentials(
+			client.credentials.awsAccessKeyID, client.credentials.awsSecretAccessKey, ""),
+	})
+	client.session = sess
+
+	return err
+}
+
+func (client *AwsClient) GetBucketName() string {
```

Comment:
not used

Reply:
Fixed in bc25eb4ad8ccd44d86b7a24694e3989552e789c7